platform/viewer/src/lib/DICOMSR/dataExchange.js

```diff
@@ -13,23 +13,37 @@ export const retrieveMeasurements = options => {
   return retrieveMeasurementFromSR(latestSeries, server);
 };
 
-export const storeMeasurements = (measurementData, filter, server) => {
+export const storeMeasurements = async (measurementData, filter, server) => {
   OHIF.log.info('[DICOMSR] storeMeasurements');
 
   if (!server || server.type !== 'dicomWeb') {
-    OHIF.log.error('DicomWeb server is required!');
+    OHIF.log.error('[DICOMSR] DicomWeb server is required!');
     return Promise.reject({});
   }
 
+  const firstMeasurementKey = Object.keys(measurementData)[0];
+  const firstMeasurement = measurementData[firstMeasurementKey][0];
   const studyInstanceUid =
-    measurementData[Object.keys(measurementData)[0]][0].studyInstanceUid;
-
-  return stowSRFromMeasurements(measurementData, server).then(
-    () => {
+    firstMeasurement && firstMeasurement.studyInstanceUid;
+
+  try {
+    const { notSupportedTools } = await stowSRFromMeasurements(
```

Comment:
`notSupportedTools` -> `unsupportedTools`

Reply:
liked it. Code is updated.